src/envs/__init__.py

```diff
@@ -51,3 +54,23 @@ def create_env(name: str) -> BaseEnv:
     if name == "repeated_nextto":
         return RepeatedNextToEnv()
     raise NotImplementedError(f"Unknown env: {name}")
+
+
+def create_env(name: str) -> BaseEnv:
+    """Create an environment given its name.
+    """
+    env = _create_new_env_instance(name)
+    _MOST_RECENT_ENV_INSTANCE[name] = env
+    return env
+
+
+def get_cached_env_instance (name: str) -> BaseEnv:
+    """Get the most recent env instance, or make a new one.
+    """
+    assert name in _MOST_RECENT_ENV_INSTANCE
+    # It is not easy to cover this because we don't control the order that
```

Comment:
it should now be possible to remove this comment and the pragma below, hopefully

Reply:
that was my secret goal the whole time :)